relay-server/src/actors/spooler/mod.rs

```diff
@@ -250,17 +264,22 @@ struct OnDisk {
     db: Pool<Sqlite>,
     buffer_guard: Arc<BufferGuard>,
     max_disk_size: usize,
+    /// The number of items spooled to disk.
+    ///
+    /// This number can be negative when relay picks up an existing db and immediately starts
+    /// unspooling.
```

Comment:
Even if `select count(*)` is expensive, I think we could consider it for startup cases, since it's only one query during startup so the overhead shouldn't be very big. We'd avoid reporting negative values in that case. What do you think?

Reply:
We should also test how fast it actually is with a completely filled database. There are conflicting statements on the internet about what the fastest way to count items is, but it seems that additional indexes or running `COUNT(_rowid_)` could be faster.

Reply:
@olksdr and I did some research & experiment on this and ruled out counting on startup, see this discussion on a previous PR: https://github.com/getsentry/relay/pull/2011#discussion_r1158649001.

I did not try `count(_rowid_)`, but since it's a linear scan either way I cannot imagine it helps us much (see https://stackoverflow.com/a/18810481). If you think it's worth it, I can investigate some more time.